include/rmm/device_buffer.hpp

```diff
@@ -383,6 +383,24 @@ class device_buffer {
    *-------------------------------------------------------------------------**/
   mr::device_memory_resource* memory_resource() const noexcept { return _mr; }
 
+  /**--------------------------------------------------------------------------*
+   * @brief Copies rmm::device_buffer to a preallocated host buffer.
+   *-------------------------------------------------------------------------**/
+  void copy_to_host(void* host_buffer, cudaStream_t stream = 0) const {
+    if (host_buffer == nullptr) {
```

Comment:
Need to call set_stream here like other functions that take a stream...

Reply:
I did it after the `nullptr` check, but happy to move it before if that is preferred.

Reply:
Sorry, missed that.

Reply:
No worries. Wasn't totally sure if that was needed in this case. So thanks for confirming 🙂